Safety to avoid modifying files outside deploy_path

hansemschnokeloch opened the pull request:
Using writable_dirs with writable_use_sudo can lead to a disaster on a server (example using /var/log instead of var/log as writable dir).

- [ ] Bug fix #…?
- [ ] New feature?
- [ ] BC breaks?
- [ ] Deprecations?
- [ ] Tests added?
- [x] Changelog updated?

      Please, update CHANGELOG.md by running next command:
      $ php bin/changelog

antonmedv commented:
Nice addition. 👍🏻

martijngastkemper commented:
Maybe add this to the docs. Because this feature removes the possibility to use absolute paths.

hansemschnokeloch commented:
I modified my draft. My first version was not safe enough, because with a space before the slash "  /var/log" the trim was not working anymore.
Now I add the deploy_path to the directories. I also added an option to disable this behavior and use absolute paths if needed.

antonmedv commented on recipe/deploy/writable.php:
```diff
@@ -39,12 +39,16 @@
 set('writable_chmod_recursive', true);
 
+// Allow absolute path in writable_dirs
+set('writable_allow_absolute_path', false);
```
I think it’s otherkill. I assume setting writable outside deploy path is really corner case and should be ignored.

antonmedv commented:
But instead trimming slashes error message to user can be shown. It’s much better design.

hansemschnokeloch commented:
I agree, throwing an error message is much better, and we need less code lines.
